fido/Crypto/Fido2/Protocol.hs

```diff
@@ -380,6 +385,35 @@ instance Aeson.ToJSON URLEncodedBase64 where
 data PublicKey = Ec2Key ECDSA.PublicKey
   deriving (Show)
 
+publicKeyPoint :: PublicKey -> ECDSA.Point
+publicKeyPoint (Ec2Key key) =
+  let ECDSA.PublicKey {public_q} = key
+   in public_q
+
+publicKeyX :: PublicKey -> Integer
+publicKeyX key =
+  case publicKeyPoint key of
+    ECDSA.Point x _y -> x
+    -- TODO: Does this ever happen?
+    ECDSA.PointO -> error "Got PointO at infinity. We always expect a key with X and Y"
+
+publicKeyY :: PublicKey -> Integer
+publicKeyY key =
+  case publicKeyPoint key of
+    ECDSA.Point _x y -> y
+    -- TODO: Does this ever happen?
+    ECDSA.PointO -> error "Got PointO at infinity. We always expect a key with X and Y"
+
+-- Make a public key at the constant curve that we currently only support.
+mkPublicKey :: Integer -> Integer -> PublicKey
```

Comment:
Shouldn’t this return `Maybe PublicKey`? What if the coordinates don’t lie on the curve?

Reply:
I've now renamed this to `mkEcdsaPublicKey` and made it check whether the coordinates are on the curve.